Code/EnginePlugins/AiPlugin/Navigation/Components/NavMeshObstacleComponent.h

```diff
@@ -0,0 +1,23 @@
+﻿#pragma once
+
+#include <AiPlugin/AiPluginDLL.h>
+#include <Core/World/Component.h>
+#include <Core/World/ComponentManager.h>
+
+using ezNavMeshObstacleComponentManager = ezComponentManager<class ezNavMeshObstacleComponent, ezBlockStorageType::Compact>;
+
+class EZ_AIPLUGIN_DLL ezNavMeshObstacleComponent : public ezComponent
```

Comment:
Please add a useful comment explaining what this component does and what a typical use case would be. Since it doesn't have any properties, it is quite mysterious and without looking at the implementation details there is no way to know what it will do.
Also it looks like you made it work specifically when it gets deactivated and reactivated. Maybe we should also just expose the InvalidateSectors() function to scripts (and make it public), so that it can be triggered at any time ?

Reply:
Yea, you're right, I wanted to make it public but forgot about it